app/controllers/MessagesController.scala

```diff
@@ -69,4 +76,26 @@ class MessagesController @Inject()(
           Future.successful(NotImplemented)
       }
   }
+
+  def getMessage(arrivalId: ArrivalId, messageId: MessageId): Action[AnyContent] = authenticateForRead(arrivalId) {
+    implicit request =>
+      if (request.arrival.messages.isDefinedAt(messageId.index) && request.arrival.messages(messageId.index).optStatus != Some(SubmissionFailed))
+        Ok(Json.toJson(ResponseMovementMessage.build(arrivalId, messageId, request.arrival.messages(messageId.index))))
+      else NotFound
+  }
+
+  def getMessages(arrivalId: ArrivalId): Action[AnyContent] = authenticateForRead(arrivalId) {
+    implicit request =>
+      Ok(
+        Json.toJson(ResponseArrival.build(
+          request.arrival,
+          request.arrival.messages.view.zipWithIndex
+            .filterNot {
+              case (message, _) => message.optStatus == Some(SubmissionFailed)
+            }
+            .map {
+              case (message, count) => ResponseMovementMessage.build(arrivalId, new MessageId(count + 1), message)
+            }
```

Comment:
I think this is logic that can be moved out of the controller and should be done by something else, possibly by whatever is doing the conversion from the Arrival to the ResponseArrival?